File: eAnimalShelter/UI/eanimalshelter_mobile/lib/models/volunteer_activity.dart
class VolunteerActivity {
  final int activityId;
  final String title;
  final String description;
  final int locationId;
  final String? locationName;
  final DateTime startDateTime;
  final DateTime endDateTime;
  final int maxVolunteers;
  final int status;
  final int currentVolunteers;
  final int applicationsCount;

  VolunteerActivity({
    required this.activityId,
    required this.title,
    required this.description,
    required this.locationId,
    required this.locationName,
    required this.startDateTime,
    required this.endDateTime,
    required this.maxVolunteers,
    required this.status,
    required this.currentVolunteers,
    required this.applicationsCount,
  });

  factory VolunteerActivity.fromJson(Map<String, dynamic> json) {
    return VolunteerActivity(
      activityId: json["activityId"],
      title: json["title"],
      description: json["description"],
      locationId: json["locationId"],
      locationName: json["locationName"],
      startDateTime: DateTime.parse(json["startDateTime"]),
      endDateTime: DateTime.parse(json["endDateTime"]),
      maxVolunteers: json["maxVolunteers"],
      status: json["status"],
      currentVolunteers: json["currentVolunteers"],
      applicationsCount: json["applicationsCount"],
    );
  }
}

File: eAnimalShelter/eAnimalShelter.Services/Database/VolunteerActivity.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using eAnimalShelter.Model.Enums;

namespace eAnimalShelter.Services.Database
{
    public class VolunteerActivity
    {
        [Key]
        public int ActivityId { get; set; }
        [Required]
        public string Title { get; set; } = string.Empty;
        [Required]
        public string Description { get; set; } = string.Empty;
        public int LocationId { get; set; }
        [ForeignKey(nameof(LocationId))]
        public Location Location { get; set; } = null!;
        public DateTime StartDateTime { get; set; }
        public DateTime EndDateTime { get; set; }
        public int MaxVolunteers { get; set; }
        public ActivityStatus Status { get; set; }
        public int CreatedBy { get; set; }
        [ForeignKey("CreatedBy")]
        public User CreatedByUser { get; set; } = null!;

        public ICollection<VolunteerAssignment> VolunteerAssignments { get; set; } = new List<VolunteerAssignment>();
    }

    
}